enhancements/console/catalog-categories.md

```diff
@@ -0,0 +1,195 @@
+
+## Motivation
+
+Customers who provide their own catalog items would like to customize the category filters such that they better represent their catalog items and make it easier for developers to navigate the catalog.
+
+### Goals
+
+- Add configuration options to allow cluster admins to customize catalog categories.
+- Provide a list of default values to the admin
+
+### Non-Goals
+
+- Hiding categories based on the available catalog items.
+- Introducing a UI to configure this categories without a yaml editor.
+
+## Proposal
+
+### Extending the Console CRD
+
+Extend the existing `operator.openshift.io/v1` / `Console` CRD. It provides already a `spec.customization` area where we can add a list of available categories.
+
+Each category could have a list of subcategories which enables the Developer Catalog to build a 2 level tree based on this data:
+
+```
+├── Customer Name
+│   ├── Project/Team A
+│   ├── Project/Team B
+│   └── ...
+├── Languages
+│   ├── Java
+│   ├── JavaScript
+│   └── ...
+├── Databases
+├── Middleware
+│   ├── Integration
+│   ├── Process Automation
+│   └── ...
+└── ...
+```
+
+Each Category and Subcategories matches following schema:
+
+- `id: string`, identifier used in the URL to enable deeplinking in console
+- `label: string`, category display label
+- `tags: string[]`, the list of item tags this category will match
+
+Categories on the first level contains also a list of subcategories:
+
+- `subcategories:` a list of child categories
+
+If the user selects a top-level category the tags of the top-level category and all subcategories are merged, so that the parent category contains all catalog items of all subcategories.
+
+So the final yaml should look like this one:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Console
+metadata:
+  annotations:
+    release.openshift.io/create-only: 'true'
+  name: cluster
+  ...
+spec:
+  customization:
+    brand: online
+    developerCatalog:
+      categories:
+      - id: languages
+        label: Languages
+        tags: [go]
+        subcategories:
+        - id: java
+          label: Java
+          tags: [java]
+        - id: javascript
+          tags: [javascript, nodejs, js]
+  ...
+```
+
+### Fetching data from ConfigMap
+
+An update to the `Console` CR will update the console deployment and update a corresponding `ConfigMap`.
+
+The console will read this customization from the `ConfigMap` "console-config" (in namespace "openshift-console").
+
+If there is any categories defined in the ConfigMap this will override the predefined (static) list of categories. The two lists are not merged.
+
+### Provide a code snippet in YAML editor
+
+The admin needs to know the current default categories to customize them. We don't want provide this defaults in the CRD so that we can change these defaults later without migrating the customer data.
```

Comment:
```suggestion
The admin needs to know the current default categories to customize them. We don't want to provide the default categories in the CR so that we can change these defaults later without migrating the customer data.
```